language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -155,15 +155,17 @@ genDefDataType curModName tpls def = case unTypeConName (dataTypeCon def) of
                 Nothing -> ((makeType typeDesc, makeSer serDesc), Set.unions fieldRefs)
                 Just tpl ->
                     let (chcs, argRefs) = unzip
-                            [((unChoiceName (chcName chc), t), argRefs)
+                            [((unChoiceName (chcName chc), t, r, rtyp), argRefs)
```

Comment:
I suppose `rtyp` is the companion object (ie. `Serializable` dictionary) for the return type. I think I've used something like `rser` below. We should find a consistent naming scheme for these. Not necessarily in this PR but soon.

Reply:
Ok